### PR TITLE
Exclude system apps in applications GET impl

### PR DESCRIPTION
**Pull request by JeethJJ**
Introduce excludeSystemPortals API query param and following implementation. 
Introduce systemPortals server level config and implementation to exclude the mentioned systemPortals in get applications api response.

### Proposed changes in this pull request
1. Add a system_portals toml config to maintain the list of system portals to be excluded when listing Applications via the API.
2. Update DB queries for the following requirements.
-  To get the count of applications excluding system portals.
- To retrieve application basic info excluding system portals.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java [hunk 1]
```diff
@@ -46,9 +46,23 @@ ApplicationBasicInfo[] getAllPaginatedApplicationBasicInfo(int pageNumber) throw
...
+     * use {@link PaginatableFilterableApplicationDAO#getApplicationBasicInfo(int, int, Boolean)}.
      */
     ApplicationBasicInfo[] getApplicationBasicInfo(int offset, int limit) throws IdentityApplicationManagementException;
 
+    /**
+     * Get all the basic application information based on the offset and the limit.
+     *
+     * @param offset Starting index of the count.
+     * @param limit  Count value.
+     * @param excludeSystemPortals  Exclude system portals.
```
Comment (shanggeeth) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java:59]: Check the formatting for parameter descriptions.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 2]
```diff
@@ -2162,6 +2181,60 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
...
+                : "";
+    }
+
+    /**
+     * Get system portals from configuration.
+     *
+     * @param parentElement
+     * @return
+     */
+    private static Set<String> getApplications(String parentElement) {
```
Comment (ZiyamSanthosh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2206]: Let's rename the method name according to its purpose.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 3]
```diff
@@ -2162,6 +2181,60 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
...
+
+        Set<String> applications = new HashSet<>();
+        while (applicationIdentifierIterator.hasNext()) {
+            OMElement applicationIdentifierConfig = (OMElement) applicationIdentifierIterator.next();
+            String applicationName = applicationIdentifierConfig.getText();
+            if (StringUtils.isNotBlank(applicationName)) {
+                applications.add(applicationName.trim());
+            }
+        }
+
```
Comment (shanggeeth) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2234]: Remove this line.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 4]
```diff
@@ -3957,6 +4030,20 @@ private String generateFilterValueForBackend(String searchOperation, String sear
...
+        return getCountOfApplications(filter, false);
+    }
+
+    /**
+     * Get count of applications for user which has the filter string
+     *
+     * @param filter
+     * @return
+     * @throws IdentityApplicationManagementException
+     */
```
Comment (shanggeeth) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:4042]: Check the method description for formatting issues.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java [hunk 5]
```diff
@@ -95,28 +95,28 @@ public class ApplicationMgtDBQueries {
...
-            " SP_APP.APP_NAME != ? AND (%s) ORDER BY SP_APP.ID DESC LIMIT ?, ?";
+            " SP_APP.APP_NAME != ? %s AND (%s) ORDER BY SP_APP.ID DESC LIMIT ?, ?";
 
     public static final String LOAD_APP_NAMES_BY_TENANT_ORACLE = "SELECT ID, APP_NAME, VERSION, DESCRIPTION, " +
             "UUID, IMAGE_URL, ACCESS_URL, USERNAME, USER_STORE, TENANT_ID FROM " +
             "(SELECT ID, APP_NAME, VERSION, DESCRIPTION, UUID, IMAGE_URL, ACCESS_URL, USERNAME, USER_STORE, " +
             "TENANT_ID, rownum AS rnum FROM " +
             "(SELECT ID, APP_NAME, VERSION, DESCRIPTION, UUID, IMAGE_URL, ACCESS_URL, USERNAME, USER_STORE, " +
-            "TENANT_ID FROM SP_APP ORDER BY ID DESC) WHERE TENANT_ID = ? AND APP_NAME != ? AND rownum <= ?) WHERE " +
+            "TENANT_ID FROM SP_APP ORDER BY ID DESC) WHERE TENANT_ID = ? AND APP_NAME != ? %s AND rownum <= ?) WHERE " +
```
Comment (shanggeeth) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java:119]: AND SP_APP.APP_NAME NOT IN (%s)might not be the correct syntax here. Please check.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java [hunk 6]
```diff
@@ -95,28 +95,28 @@ public class ApplicationMgtDBQueries {
...
+            "AND SP_APP.APP_NAME != ? %s AND (%s)";
     public static final String LOAD_APP_BY_TENANT_AND_NAME = "SELECT ID, APP_NAME, VERSION, DESCRIPTION, UUID, " +
             "IMAGE_URL, ACCESS_URL, IS_DISCOVERABLE, USERNAME, USER_STORE, TENANT_ID FROM SP_APP WHERE " +
             "TENANT_ID = :TENANT_ID; AND APP_NAME = :APP_NAME;";
 
     // Load application basic information for listing with pagination
     public static final String LOAD_APP_NAMES_BY_TENANT_MYSQL = "SELECT ID, APP_NAME, VERSION, DESCRIPTION, " +
             "UUID, IMAGE_URL, ACCESS_URL, USERNAME, USER_STORE, TENANT_ID FROM SP_APP WHERE TENANT_ID = ? AND " +
-            "APP_NAME != ? ORDER BY ID DESC LIMIT ?, ?";
+            "APP_NAME != ? %s ORDER BY ID DESC LIMIT ?, ?";
```
Comment (shanggeeth) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java:106]: `SPAPP` part in `AND SP_APP.APP_NAME NOT IN (%s)` might not be the correct syntax here. Please check.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java (hunk 1 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/964c6d3814dba4a93591240fb7b95b4f3b9f77c3

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 2 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/964c6d3814dba4a93591240fb7b95b4f3b9f77c3

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 3 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/964c6d3814dba4a93591240fb7b95b4f3b9f77c3

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 4 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/964c6d3814dba4a93591240fb7b95b4f3b9f77c3

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java [hunk 7]
```diff
@@ -127,13 +127,13 @@ public class ApplicationMgtDBQueries {
...
-            "WHERE SP_APP.TENANT_ID = ? AND SP_APP.APP_NAME <> ? AND (%s) ORDER BY SP_APP.ID DESC) " +
+            "WHERE SP_APP.TENANT_ID = ? AND SP_APP.APP_NAME <> ? %s AND (%s) ORDER BY SP_APP.ID DESC) " +
             "WHERE ROWNUM <= ?) WHERE rn > ?";
 
     public static final String LOAD_APP_NAMES_BY_TENANT_DB2SQL = "SELECT ID, APP_NAME, VERSION, DESCRIPTION, " +
             "UUID, IMAGE_URL, ACCESS_URL, USERNAME, USER_STORE, TENANT_ID FROM (SELECT " +
-            "ROW_NUMBER() OVER(ORDER BY ID DESC) AS rn,SP_APP.* FROM SP_APP WHERE TENANT_ID = ? AND APP_NAME != ? ) " +
-            "WHERE rn BETWEEN ? AND ?";
+            "ROW_NUMBER() OVER(ORDER BY ID DESC) AS rn,SP_APP.* FROM SP_APP WHERE TENANT_ID = ? " +
+            "AND APP_NAME != ? %s ) WHERE rn BETWEEN ? AND ?";
```
[the remark this answers is not in the document]
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java:136]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/390d2fe79c6ef33c9fb2b82d0c545d835823a303

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java (hunk 5 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java:119]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/390d2fe79c6ef33c9fb2b82d0c545d835823a303

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java (hunk 6 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java:106]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/390d2fe79c6ef33c9fb2b82d0c545d835823a303

**Review by shanggeeth — DISMISSED** [review]

**Comment by shanggeeth** [comment]
Let's run the PR builder before merging.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java [hunk 8]
```diff
@@ -94,10 +94,28 @@ ApplicationBasicInfo[] getPaginatedApplicationBasicInfo(String tenantDomain, Str
...
+     * @param tenantDomain          Tenant Domain.
+     * @param username              User name.
+     * @param filter                Application name filter.
+     * @param offset                Starting index of the count.
+     * @param limit                 Counting value.
+     * @param excludeSystemPortals  Exclude system portals.
+     * @return An array of {@link ApplicationBasicInfo} instances matching the given filter within the given limit.
+     * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     */
+    ApplicationBasicInfo[] getApplicationBasicInfo(String tenantDomain, String username, String filter, int offset,
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java:115]: Should be a default method.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java [hunk 9]
```diff
@@ -117,8 +135,23 @@ int getCountOfAllApplications(String tenantDomain, String username)
...
+     * Get count of all basic application information for a matching filter.
+     *
+     * @param tenantDomain          Tenant Domain.
+     * @param username              User Name.
+     * @param filter                Application name filter.
+     * @param excludeSystemPortals  Exclude system portals.
+     * @return int Count of applications.
+     * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
+     */
+    int getCountOfApplications(String tenantDomain, String username, String filter, Boolean excludeSystemPortals)
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java:154]: Should be a default method.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java [hunk 10]
```diff
@@ -471,6 +471,14 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String tenantDomain, Strin
...
 
+        return getApplicationBasicInfo(tenantDomain, username, filter, offset, limit, false);
+    }
+
+    @Override
+    public ApplicationBasicInfo[] getApplicationBasicInfo(String tenantDomain, String username, String filter,
+                                                          int offset, int limit, Boolean excludeSystemPortals)
+            throws IdentityApplicationManagementException {
+
         return new ApplicationBasicInfo[0];
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java:482]: Can remove from here and move to the default method

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java [hunk 11]
```diff
@@ -94,10 +94,28 @@ ApplicationBasicInfo[] getPaginatedApplicationBasicInfo(String tenantDomain, Str
      * @param limit        Counting value.
      * @return An array of {@link ApplicationBasicInfo} instances matching the given filter within the given limit.
      * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     * @Deprecated This is being deprecated as introducing excludeSystemPortals to exclude system portals in response
+     * use {@link ApplicationPaginationAndSearching#getApplicationBasicInfo(String, String, String, int, int, Boolean)}.
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java:98]: Add the deprecation tag

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java [hunk 12]
```diff
@@ -46,9 +46,23 @@ ApplicationBasicInfo[] getAllPaginatedApplicationBasicInfo(int pageNumber) throw
...
+    /**
+     * Get all the basic application information based on the offset and the limit.
+     *
+     * @param offset                Starting index of the count.
+     * @param limit                 Count value.
+     * @param excludeSystemPortals  Exclude system portals.
+     * @return An array of {@link ApplicationBasicInfo} instances within the limit.
+     * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     */
+    ApplicationBasicInfo[] getApplicationBasicInfo(int offset, int limit, Boolean excludeSystemPortals)
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java:63]: Should be a default method. Check for other places as well.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java [hunk 13]
```diff
@@ -46,9 +46,23 @@ ApplicationBasicInfo[] getAllPaginatedApplicationBasicInfo(int pageNumber) throw
      * @param limit  Count value.
      * @return An array of {@link ApplicationBasicInfo} instances within the limit.
      * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     * @Deprecated This is being deprecated as introducing excludeSystemPortals to exclude system portals in response.
+     * use {@link PaginatableFilterableApplicationDAO#getApplicationBasicInfo(int, int, Boolean)}.
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java:50]: Add the deprecation annotation. Check for other places as well.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 14]
```diff
@@ -2162,6 +2181,66 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
...
+            String excludeSystemPortalsQuery = useTableNameInQuery ?
+                    ApplicationMgtDBQueries.EXCLUDE_SYSTEM_PORTALS_BY_TABLE_NAME_AND_NAME :
+                    ApplicationMgtDBQueries.EXCLUDE_SYSTEM_PORTALS_BY_NAME;
+            Set<String> systemPortals = getSystemPortalsFromConfiguration(SYSTEM_PORTALS);
+            return systemPortals.isEmpty() ? "" :
+                    String.format(excludeSystemPortalsQuery, systemPortals.stream()
+                                    .map(s -> "'" + s + "'")
+                                    .collect(Collectors.joining(", ")));
+        } else {
+            return "";
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2203]: Lets update other places as well.
```suggestion
            return StringUtils.EMPTY;
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 15]
```diff
@@ -2162,6 +2181,66 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
...
+        }
+    }
+
+    /**
+     * Get system portals from configuration.
+     *
+     * @param parentElement Configuration tag name.
+     * @return Set of system portals from default.json.
+     */
+    private static Set<String> getSystemPortalsFromConfiguration(String parentElement) {
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2213]: Should be able to use IdentityUtil.getPropertyAsList instead

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 16]
```diff
@@ -2074,8 +2079,16 @@ public ApplicationBasicInfo[] getPaginatedApplicationBasicInfo(int pageNumber, S
     public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset, int limit)
             throws IdentityApplicationManagementException {
 
+        return getApplicationBasicInfo(filter, offset, limit, false);
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2082]: Mark as deprecated

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java [hunk 17]
```diff
@@ -331,10 +339,18 @@ public ApplicationBasicInfo[] getPaginatedApplicationBasicInfo(int pageNumber, S
     @Override
     public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset, int limit)
             throws IdentityApplicationManagementException {
+        return getApplicationBasicInfo(filter, offset, limit, false);
```
Comment (janakamarasena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java:342]: Add new line

**Comment by janakamarasena** [comment]
Make sure to update the integration tests cover the new cases.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 16 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2079]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java (hunk 17 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java:345]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java (hunk 8 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java (hunk 9 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java (hunk 12 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/PaginatableFilterableApplicationDAO.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 15 above)
Comment (JeethJJ) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**